### PR TITLE
Fixes #9154

### LINKED ISSUE #9154: `nim js -o:dirname main.nim` writes nothing, and no error shown
mkdir dirname
nim js -o:dirname main.nim writes nothing, and no error shown

it should either write to dirname/main.js, or error with : cannot write to dirname (dirname is a directory)

### PR DESCRIPTION
**Pull request by manterolat**
Add checks to frontend output argument:
 - The output file actually being a directory
 - A nonexistent path to the output file

### CODE REVIEW

**Comment by manterolat** [comment]
Seems like this PR found a bug in `kochdocs.nim`, where the `web/upload` folder is used before it's created. Should I make a separate branch/pull request to fix this?

**Comment by Araq** [comment]
This is not the way to do it, the compiler should create the required output directories instead.

File: compiler/cgen.nim [hunk 1]
```diff
@@ -1630,6 +1630,9 @@ proc cgenWriteModules*(backend: RootRef, config: ConfigRef) =
   # deps are allowed (and the system module is processed in the wrong
   # order anyway)
   g.config = config
+  let (outDir, _, _) = splitFile(config.outfile)
+  if not outDir.isEmpty:
+    try: createDir(outDir) except: discard "couldn't create dir"
```
Comment (Araq) [inline compiler/cgen.nim]: Let the exception bubble up, don't hide it.

File: compiler/jsgen.nim [hunk 2]
```diff
@@ -2282,6 +2282,9 @@ proc myClose(graph: ModuleGraph; b: PPassContext, n: PNode): PNode =
         else: AbsoluteFile(getCurrentDir() / m.config.outFile.string)
       else:
         changeFileExt(completeCFilePath(m.config, AbsoluteFile f), ext)
+    let (outDir, _, _) = splitFile(outfile)
+    if not outDir.isEmpty:
+      try: createDir(outDir) except: discard "couldn't create dir"
```
Comment (Araq) [inline compiler/jsgen.nim]: Let the exception bubble up, don't hide it.

**Pull request merged**